error_writer.go

```diff
@@ -54,69 +52,54 @@ type ErrorWriter struct {
 // Options supplied via [WithConditionalHandlerOptions] are ignored.
 func NewErrorWriter(opts ...HandlerOption) *ErrorWriter {
 	config := newHandlerConfig("", StreamTypeUnary, opts)
-	writer := &ErrorWriter{
+	codecs := newReadOnlyCodecs(config.Codecs)
```

Comment:
The GoDoc for `NewErrorWriter` also needs to be updated to explain why we need `...HandlerOption`: now it's not for codecs, it's for the Connect version header.